### PR TITLE
feat: optimize node_modules bundle size and unused code

### PR DESCRIPTION
**Pull request by capJavert**
## Changes

- remove lazysizes - from what I can see we no longer use this in any way and prefer native lazy loading)
- remove cross-fetch - it is a hard dependency and included from graphql-request, I manually remove it from bundle in webpack and patch graphql-request to allow it, same was done in newer version of graphql-request but can't update now because of bunch of breaking changes)
- lazy init react modal - the whole react-modal package is included in the main bundle due to import in _app, I added lazy init inside `Modal` component (left companion/extension as is)
- remove lodash.shuffle - it became quite big, replaced with same implementation
- remove lodash.clonedeep - replaced with structuredClone

From my testing everything seems to work fine but would appreciate sanity checks 🙏 

Decrease of about 15% of main bundle size and about 10-15kb flat for each page bundle.

Before (from next build):
```
+ First Load JS shared by all              224 kB
  ├ chunks/framework-c6c82aad00023883.js   44.9 kB
  ├ chunks/main-73b80513afb42ad1.js        39.7 kB
  ├ chunks/pages/_app-c7d8959a0c0bc8e9.js  96.8 kB
  ├ css/6d697062275c3f9d.css               33.7 kB
  └ other shared chunks (total)            9.31 kB
```

After:
```
+ First Load JS shared by all              209 kB ✅
  ├ chunks/framework-c6c82aad00023883.js   44.9 kB
  ├ chunks/main-73b80513afb42ad1.js        39.7 kB
  ├ chunks/pages/_app-873f249519293e75.js  81.6 kB ✅
  ├ css/6d697062275c3f9d.css               33.7 kB
  └ other shared chunks (total)            9.22 kB ✅
```

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

WT-2222

### Preview domain

https://wt-2222-node-modules.preview.app.daily.dev/

### CODE REVIEW

File: packages/shared/package.json [hunk 1]
```diff
@@ -128,7 +124,6 @@
     "webextension-polyfill-ts": "^0.22.0"
   },
   "sideEffects": [
-    "./src/lib/lazysizesImport.ts",
```
Comment (rebelchris) [inline packages/shared/package.json:131]: Yeah legacy I think, probably missed the cleanup somewhere.

**Review by rebelchris — APPROVED** [review]
Nice work!
Pretty good result for such minimal implications.

This is why I often prefer internal implementations of existing hooks over importing package for all the overhead.
Wonder if we can do it with some more perhaps?

(If tree shaking actually would work it'd even be more impressive probably)

**Review by nensidosari — APPROVED** [review]
🚀

**Comment by nensidosari** [comment]
Will just do some sanity checks now :)

**Comment by nensidosari** [comment]
Looks good from my tests

**Review by AmarTrebinjac — APPROVED** [review]

**Pull request merged**